conans/model/conan_generator.py

```diff
@@ -43,5 +51,48 @@ def content(self):
     def filename(self):
         raise NotImplementedError()
 
+
+class GeneratorComponentsMixin(object):
```

Comment:
why not adding this to the `Generator` class?

Reply:
Just to keep things isolated. If you have the components feature (`cmake_find_package` and `pkg_config`) you add this class to your inheritance and then you get some new powers/overrides for your functions. In the next PR you  can see an example here: https://github.com/conan-io/conan/pull/7644/files#diff-496c84b81260eff9ea669a196e9054dbR104, you get the new implementation of the `get_public_deps` only if you add the mixin to your parent classes. 

Usually (not the best example here) this avoids and extra `if/else` and reduces the [complexity](https://en.wikipedia.org/wiki/Cyclomatic_complexity) of the function and the number of unittests required (type/inheritance is hardcoded at _compile time_, while `if/else` is resolved at runtime).